File: di/dagger3/src/main/java/com/example/dagger3/MainActivity.java
package com.example.dagger3;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.example.dagger3.data.CarRepository;
import com.example.dagger3.di.HomeActivityComponent;

import jakarta.inject.Inject;

public class MainActivity extends AppCompatActivity {
    HomeActivityComponent mComponent;

    @Inject
    Fragment mHomeFragment;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mComponent = ((CarApp) (getApplication())).appComponent()
                .homeActivityComponent().build();
        mComponent.inject(this);

        setContentView(R.layout.activity_main);
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.container_home, mHomeFragment)
                .commit();
    }

    public HomeActivityComponent getComponent() {
        return mComponent;
    }
}

File: app/src/main/java/com/xxh/learn/sample/MainActivity.kt
package com.xxh.learn.sample

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.createGraph
import androidx.navigation.fragment.NavHostFragment
import com.xxh.learn.sample.nav.NavDestinations
import com.xxh.learn.sample.nav.navGraph


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initNav()
    }

    private fun initNav() {
        val navHostFragment =
            supportFragmentManager.findFragmentById(R.id.fragmentContainerView_main) as NavHostFragment
        val navController = navHostFragment.navController
        navController.graph = navController.createGraph(
            startDestination = NavDestinations.Main,
        ) {
            navGraph()
        }
    }
}
